test/metabase/api/setup_test.clj

```diff
@@ -92,25 +93,32 @@
 
 (deftest invite-user-test
   (testing "POST /api/setup"
-    (testing "Check that a second admin can be created during setup, and that an invite email is sent successfully"
+    (testing "Check that a second admin can be created during setup, and that an invite email is sent successfully and
+             a Snowplow analytics event is sent"
       (mt/with-fake-inbox
-        (let [email (mt/random-email)
-              first-name (mt/random-name)
-              last-name (mt/random-name)
-              invitor-first-name (mt/random-name)]
-          (with-setup {:invite {:email email, :first_name first-name, :last_name last-name}
-                       :user {:first_name invitor-first-name}
-                       :site_name "Metabase"}
-            (let [invited-user (User :email email)]
-              (is (= (:first_name invited-user) first-name))
-              (is (= (:last_name invited-user) last-name))
-              (is (:is_superuser invited-user)))
-            (let [invite-email (-> (mt/regex-email-bodies
-                                    (re-pattern (str invitor-first-name " could use your help setting up Metabase.*")))
-                                   (get email)
-                                   first)]
-             (is (= {(str invitor-first-name " could use your help setting up Metabase.*") true}
-                    (:body invite-email))))))))
+        (snowplow-test/with-fake-snowplow-collector
+          (let [email (mt/random-email)
+                first-name (mt/random-name)
+                last-name (mt/random-name)
+                invitor-first-name (mt/random-name)]
+            (with-setup {:invite {:email email, :first_name first-name, :last_name last-name}
+                         :user {:first_name invitor-first-name}
+                         :site_name "Metabase"}
+              (let [invited-user (User :email email)]
+                (is (= (:first_name invited-user) first-name))
+                (is (= (:last_name invited-user) last-name))
+                (is (:is_superuser invited-user))
+                (is (partial= [{:data {"event"           "invite_sent",
+                                       "invited_user_id" (u/the-id invited-user)
+                                       "source"          "setup"}}]
+                              (filter #(= (get-in % [:data "event"]) "invite_sent")
+                                      (snowplow-test/pop-event-data-and-user-id!)))))
+              (let [invite-email (-> (mt/regex-email-bodies
+                                      (re-pattern (str invitor-first-name " could use your help setting up Metabase.*")))
+                                     (get email)
+                                     first)]
+                (is (= {(str invitor-first-name " could use your help setting up Metabase.*") true}
```

Comment:
There are two new email checking functions in `metabase.email-test` and exposted in `mt`:
- `(received-email-subject? :rasta #"some subject")`
- `(received-email-body? :rasta #"something in the body")`

These test that there is an email sent to the user matching the regex either against the body or the subject of the email. It would probably be better than this even though its pretty similar. Its the actual predicate we normally care about.